Bio/SeqIO/pepdigestIO.py

```diff
@@ -0,0 +1,143 @@
+        # Reads the main header
+        header_sep_count = 0
+        i = 0
+        while header_sep_count < 2:
+            line = handle.readline()
+            if self.main_header_pttrn.search(line) is not None:
+                header_sep_count += 1
+            i += 1
+            if i > self.max_iter:
+                raise ValueError("Broken header in start of file.")
+
+        records = self.iterate(handle)
+
+        return records
+
+    def parse_digestion_header(self, handle):
+        """Parse the header of a digestion.
+
+        Return usefull information for parsing the results.
+        """
+        # Read sequence name and positions
+        i = 0
+        line = handle.readline()
+        while not line.startswith("# Sequence"):
+            line = handle.readline()
+            i += 1
+            if i > self.max_iter:
+                raise ValueError(
+                    "Can't find '# Sequence' section in digestion results header"
+                )
+
+        line = line.rsplit("# Sequence:")[1].strip()
+        orig_seq_name, line = line.split(None, 1)
+        line, seq_to = line.split("to:")
+        seq_to = int(seq_to)
+        seq_from = int(line.replace("from:", ""))
+
+        # Read HitCount
+        line = handle.readline()
+        hitcount = int(line.replace("# HitCount:", ""))
+
+        # Read enzyme utilized in digestion
+        handle.readline()
+        line = handle.readline()
+        enzyme = line.split("with")[1].split("yields")[0].strip()
+
+        # Read end of header
+        i = 0
+        while self.digestion_title_pttrn.search(line) is None:
+            line = handle.readline()
+            i += 1
+            if i > self.max_iter:
+                raise ValueError("Malformed digestion results header")
+
+        return orig_seq_name, seq_to, seq_from, hitcount, enzyme
+
+    def parse_digestion_results(self, handle, header):
+        """Parse the digested peptides."""
+        orig_seq_name, seq_to, seq_from, hitcount, enzyme = header
+        # skip one line
+        handle.readline()
+
+        header = handle.readline().split()
+        for i in range(hitcount):
+            seq_line = handle.readline()
+            start, end, mol_w, cterm, nterm, seq = seq_line.split()
+
+            pep_name = f"{orig_seq_name}_{i+1}"
+            yield SeqRecord(
+                Seq(seq),
+                id=pep_name,
+                name=pep_name,
+                annotations={
+                    "Start": int(start),
+                    "End": int(end),
+                    "Mol_W": float(mol_w),
+                    "Cterm": cterm,
+                    "Nterm": nterm,
+                    "source_sequence": orig_seq_name,
+                    "source_seq_from": seq_from,
+                    "source_seq_to": seq_to,
+                    "digestion_enzyme": enzyme,
+                },
+                description="generated from pepdigest",
+            )
+
+    def iterate(self, handle):
```

Comment:
Why not include this in the ``.parse()`` method?

Reply:
Don't remeber my reasoning, will do